File: solutions/1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.js
var check = function(nums) {
    let breaks = 0;
    const n = nums.length;

    for (let i = 0; i < n; i++) {
        if (nums[i] > nums[(i + 1) % n]) {
            breaks++;
        }
    }

    return breaks <= 1;
};

File: solutions/1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.rs
impl Solution {
    pub fn check(nums: Vec<i32>) -> bool {
        let n = nums.len();
        let mut breaks = 0;

        for i in 0..n {
            if nums[i] > nums[(i + 1) % n] {
                breaks += 1;
            }
        }

        breaks <= 1
    }
}